### ISSUE TITLE
Add Trainer pipeline

### ISSUE THREAD

**Issue by davidmezzetti** [neuml/txtai#96]
Add a pipeline to train new Hugging Face models using the Trainer API.

This pipeline will support the following:

- Building transient models without requiring an output directory
- Load training data from Hugging Face datasets, pandas DataFrames and list of dicts
- Text sequence classification tasks (single/multi label classification and regression) including all GLUE tasks
- All TrainingArguments passed to trainer

Some examples of how the functionality will work:

```python
import pandas as pd

from datasets import load_dataset

from txtai.pipeline import HFTrainer

trainer = HFTrainer()

# Pandas DataFrame
df = pd.read_csv("training.csv")
model, tokenizer = trainer("bert-base-uncased", df)

# Hugging Face dataset
ds = load_dataset("glue", "sst2")
model, tokenizer = trainer("bert-base-uncased", ds["train"])

# List of dicts
dt = [{"text": "sentence 1", "label": 0}, {"text": "sentence 2", "label": 1}]]
model, tokenizer = trainer("bert-base-uncased", dt)

# Support additional TrainingArguments
model, tokenizer = trainer("bert-base-uncased", dt, learning_rate=3e-5, num_train_epochs=5)
```

An full-featured example on how to use the Trainer API can be found here - https://github.com/huggingface/transformers/blob/master/examples/pytorch/text-classification/run_glue.py
